src/LibraryManager/Manifest.cs

```diff
@@ -121,12 +121,295 @@ private static bool IsValidManifestVersion(string version)
+                string providerId = string.IsNullOrEmpty(state.ProviderId) ? DefaultProvider : state.ProviderId;
+                IProvider provider = _dependencies.GetProvider(providerId);
+
+                if (provider == null)
+                {
+                    continue;
+                }
+
+                if (!providerCatalog.ContainsKey(provider))
+                {
+                    ILibraryCatalog catalog = provider.GetCatalog();
+                    if (catalog == null)
+                    {
+                        continue;
+                    }
+                    providerCatalog[provider] = catalog;
+                }
+
+                ILibrary library = await providerCatalog[provider].GetLibraryAsync(state.LibraryId, cancellationToken);
+                string libraryId = library?.Name ?? state.LibraryId;
+
+                string key = libraryId + providerId + installDestination;
+                if (!libraryToIdMap.ContainsKey(key))
+                {
+                    libraryToIdMap[key] = new List<ILibraryInstallationState>();
+                }
+
+                libraryToIdMap[key].Add(state);
+            }
+
+            return libraryToIdMap.Where(l => l.Value.Count > 1).Select(list => list.Value);
+        }
+
+        private bool CheckAlreadyInstalled(LibraryInstallationState desiredState)
+        {
+            // compare id, provider and destination
+            return Libraries.Any(l => l.LibraryId == desiredState.LibraryId
+                        && (l.DestinationPath == desiredState.DestinationPath)
+                        && (l.ProviderId == desiredState.ProviderId 
+                            || (l.ProviderId == null && desiredState.ProviderId == DefaultProvider)));
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="libraryToUpdate"></param>
+        /// <param name="newId"></param>
+        /// <param name="deleteFileAction"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<ILibraryInstallationResult> UpdateLibraryAsync(
+            ILibraryInstallationState libraryToUpdate,
+            string newId,
+            Action<string> deleteFileAction,
+            CancellationToken cancellationToken)
+        {
+            return await UpdateLibraryInternalAsync(libraryToUpdate, newId, true, deleteFileAction, cancellationToken);
+        }
+
+        /// <summary>
+        /// Updates a given library to the latest version
+        /// </summary>
+        /// <param name="libraryToUpdate">The library to update</param>
+        /// <param name="usePreRelease">Should prelease versions be considered</param>
+        /// <param name="deleteFileAction"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<ILibraryInstallationResult> UpdateLibraryToLatestAsync(
```

Comment:
You shouldn't have to implement all that. This should follow the same code path for the Update command from the lightbulb in the editor. Find out the replacement, and call install on it. Don't worry about validation at this point. If anything goes wrong, we revert to original state once we have it implemented..

Reply:
I looked at that code. That flow cannot be used to for command line scenarios. 
The VS command basically allows moving to a invalid manifest state, which deletes the files first and then fails at restoring the new one. 
(Change libraryId from jquery@3.3.1 to jquery@0.0.0)
Since in VS you can undo, its fine for the user to manually do this, but on commandline, we cannot delete files first and then tell the user 'oops we cannot install the new one'.